README.md

```diff
@@ -73,24 +56,221 @@ The `-v /path/to/your/local/models:/workspace/models` part will make sure
 the downloaded models are cached to your host storage. 
 which makes rerunning stuff much faster.
 
-The `-v ${PWD}:/workspace/spacer/` mounts your current folder including 
-`secrets.json` so that the container has the right permissions.
+The `-v ${PWD}:/workspace/spacer/` mounts your current folder (including 
+`secrets.json`, if used) so that the container has the right permissions.
 
 The last step will run the default CMD command specified in the dockerfile 
-(unit-test with coverage). If you want to enter the docker container 
+(unit-test with coverage). If you want to enter the docker container, 
 run the same command but append `bash` in the end.
 
-#### Pip install
+### Pip install
 * `pip install pyspacer`
-* Set environmental variables.
+* Set up configuration. `secrets.json` isn't an option here, so either use environment variables, or Django settings if you have a Django project.
 
-#### Local clone
+### Local clone
 * Clone this repo
+  * If using Windows: turn Git's `autocrlf` setting off before your initial checkout. Otherwise, pickled classifiers in `spacer/tests/fixtures` will get checked out with `\r\n` newlines, and the pickle module will fail to load them, leading to test failures. However, autocrlf should be left on when adding any new non-pickle files.
 * `pip install -r requirements.txt`
+* Set up configuration
+
+
+## Code overview
+
+Spacer executes tasks as defined in messages. The message types are defined
+in `messages.py` and the tasks in `tasks.py`. Several data types which can be used for input and output serialization are defined
+in `data_classes.py`.
+
+For examples on how to create spacer tasks, refer to the Core API section below, and the unit tests in `test_tasks.py`.
+
+Tasks can be executed directly by calling the methods in tasks.py. 
+However, spacer also supports an interface with AWS Batch 
+handled by `env_job()` in `mailman.py`. 
```

Comment:
Good for now, but as discussed I think decoupling should mean that no particular interface is privileged -- I should be able to invoke any of the tasks via whatever cloud system is desired. My take is, that should be via CLI and `argparse` so that the container could be run directly.

Reply:
Agreed, so removing AWS Batch specifics would at least be a "finish decoupling" bullet point. CoralNet's AWS Batch workflow could be described as an example, but not tied to pyspacer on a deeper level than that.